core/src/main/java/org/mule/runtime/core/api/MuleMessage.java

```diff
@@ -113,7 +114,13 @@ static MuleMessage of(Object payload)
         Builder payload(Object payload);
 
         @Override
+        CollectionBuilder collectionPayload(Iterator payload, Class<?> itemType);
+
+        @Override
         CollectionBuilder collectionPayload(Collection payload, Class<?> itemType);
+
+        @Override
+        CollectionBuilder collectionPayload(Object[] payload, Class<?> itemType);
```

Comment:
Do we need this, given the type can be obtained via payload.getClass().getComponentType()?  Can't we just add logic to create a CollectionDataType if `payload(String[] stringArray)` is used?